### DISCUSSION TITLE
Package or deploy serverless via CLI by specifying which .env file to use.

### DISCUSSION THREAD

**Discussion by bs-thomas** [serverless/serverless#11166] [Q&A] [answered]
May I ask if it is currently possible to package or deploy serverless via CLI by specifying which .env file to use?  It would be nice to have if this is not yet possible.

For instance, something like this:
`serverless deploy --aws-profile "some@cool.profile" --stage="dev" --region="us-east-1" --env=".env.www.cool-website.com.dev"`

**Comment by pgrzesik** [comment, marked as answer]
Hey @bs-thomas - at the moment it's not possible - the `.env` file is loaded, as well as `.env.<stage>` file if found - so you can use stage-specific `.env` files, maybe that will solve your problem?

**Reply by bs-thomas** [reply]
@pgrzesik Thank you very much for your prompt and kind assistance.  I do understand that there is a .env.<stage>, but as you can see in my example, I actually have multiple websites, and each website will have different environements.  They will all share the same serverless.yml.  So the <stage> trick is probably not going to work for me.

However, I'm thinking of just creating a script to copy the correct .env file to the root directory before calling sls deploy.  But I think it'd be a "nice-to-have" feature to allow specification of the config file via CLI, as it sounds like it can be pretty useful for users that work outside of the usual directory structure.

**Reply by pgrzesik** [reply]
I see, so unfortunately differentiating by stage is not possible in your case. The suggested workaround would obviously work, but it's also not a great user experience. We wanted to keep the dotenv support fairly simple, but we can consider expanding it a bit to accomodate such use cases as yours in the future.
